site/src/job_queue/mod.rs

```diff
@@ -174,12 +178,63 @@ pub async fn build_queue(
     Ok(queue)
 }
 
+/// From a benchmark_request create all the required jobs
+pub fn create_benchmark_jobs(
+    benchmark_request: &BenchmarkRequest,
+) -> anyhow::Result<Vec<BenchmarkJob>> {
+    anyhow::ensure!(
+        benchmark_request.tag().is_some(),
+        "Benchmark request has no tag"
+    );
+
+    let backends = benchmark_request.backends()?;
+    let profiles = benchmark_request.profiles()?;
+    let mut jobs = vec![];
+
+    // Target x benchmark_set x backend x profile -> BenchmarkJob
+    for target in Target::all() {
+        for benchmark_set in 0..benchmark_set_count(
+            collector::compile::benchmark::target::Target::from_db_target(&target),
+        ) {
+            for backend in backends.iter() {
+                for profile in profiles.iter() {
+                    let job = BenchmarkJob::new(
```

Comment:
Similar to benchmark requests, I'd like to only use public API to interact with jobs, and avoid giving everyone the option to create invalid states. So ideally the `new` function shouldn't take status and `created_at`; it should automatically set `created_at` to now (unless we *really* need to set the time in tests), and it should automatically set the status to queued.

The `database` crate should be the only one who will be able to create the job from arbitrary state, because it needs to do it for deserialization from the DB.